nixos/modules/security/doas.nix

```diff
@@ -262,31 +268,30 @@ in
     };
 
     environment.etc."doas.conf" = {
-      source = pkgs.runCommand "doas-conf"
-        {
-          src = pkgs.writeText "doas-conf-in" ''
-            # To modify this file, set the NixOS options
-            # `security.doas.extraRules` or `security.doas.extraConfig`. To
-            # completely replace the contents of this file, use
-            # `environment.etc."doas.conf"`.
-
-            # extraRules
-            ${concatStringsSep "\n" (lists.flatten (map mkRule cfg.extraRules))}
-
-            # extraConfig
-            ${cfg.extraConfig}
-
-            # "root" is allowed to do anything.
-            permit nopass keepenv root
-          '';
-          preferLocalBuild = true;
-        }
+      source = pkgs.runCommand "doas-conf" {
+        src = pkgs.writeText "doas-conf-in" ''
+          # To modify this file, set the NixOS options `security.doas.extraRules` or `security.doas.extraConfig`.
+          # To completely replace the contents of this file, use `environment.etc."doas.conf"`.
+
+          # extraRules
+          ${concatStringsSep "\n" (lists.flatten (map mkRule cfg.extraRules))}
+
+          # extraConfig
+          ${cfg.extraConfig}
+
+          # "root" is allowed to do anything.
+          permit nopass keepenv root
+        '';
+        nativeBuildInputs = [ cfg.package ];
+        preferLocalBuild = true;
+      } ''
         # Make sure that the doas.conf file is syntactically valid.
-        "${pkgs.buildPackages.doas}/bin/doas -C $src && cp $src $out";
+        doas -C $src
+        cp $src $out
```

Comment:
This script doesn't have `-e` set, so this will say that a config file "validated" even if it didn't. Please go back to the `&& cp`, which is easier to reason about.